### ISSUE TITLE
Ice/metrics objective-c failure

### ISSUE THREAD

**Issue by pepone** [zeroc-ice/ice#568]
```
Running on ice-dist-macos-mojave-test-3.7
*** [19/30] Running objective-c/Ice/metrics tests ***
[ running client/server test - 10/03/19 08:23:55 ]
- Config: xcodesdk,iphonesimulator,tcp
(executing `Ice/metrics/server' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Default.Host=127.0.0.1', '--Ice.Warn.Connections=1', '--Ice.IPv6=0', '--Ice.ThreadPool.Server.SizeMax=3', '--Ice.ThreadPool.Server.Size=1', '--Ice.Default.Protocol=tcp', '--Ice.ThreadPool.Server.SizeWarn=0'])
(executing `Ice/metrics/client' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Default.Host=127.0.0.1', '--Ice.Warn.Connections=1', '--Ice.IPv6=0', '--Ice.Default.Protocol=tcp'])
testing metrics admin facet checkedCast... ok
testing group by none...ok
testing connection metrics... ok
testing connection establishment metrics... ok
testing dispatch metrics... ok
testing invocation metrics... ok
testing metrics view enable/disable...ok
[ running client/server test - 10/03/19 08:23:57 ]
- Config: xcodesdk,iphonesimulator,ssl,serialize,mx
(executing `Ice/metrics/server' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Warn.Connections=1', '--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceSSL.Password=password', '--IceSSL.Keychain=server.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Admin.InstanceName=Server', '--Ice.ThreadPool.Server.Size=1', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--Ice.ThreadPool.Server.SizeMax=3', '--IceSSL.CertFile=server.p12', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice-dist/3.7/dist-utils/build/ice/builds/ice-clang-default/certs', '--IceSSL.KeychainPassword=password', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--IceSSL.CAs=cacert.pem', '--Ice.ThreadPool.Server.SizeWarn=0'])
(executing `Ice/metrics/client' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Warn.Connections=1', '--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceSSL.Password=password', '--IceSSL.Keychain=client.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Admin.InstanceName=Client', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--IceSSL.CertFile=client.p12', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice-dist/3.7/dist-utils/build/ice/builds/ice-clang-default/certs', '--IceSSL.KeychainPassword=password', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--IceSSL.CAs=cacert.pem'])
testing metrics admin facet checkedCast... ok
testing group by none...ok
testing connection metrics... ok
testing connection establishment metrics... ok
testing dispatch metrics... ok
testing invocation metrics... failed!
/Users/vagrant/workspace/ice-dist/3.7/dist-utils/build/ice/builds/ice-clang-default/objective-c/test/Ice/metrics/AllTests.m:1082: assertion `rim1.current == 0 && rim1.total == 6 && rim1.failures == 6' failed
Traceback (most recent call last):
  File "../scripts/Util.py", line 1700, in run
    self.runWithDriver(current)
  File "../scripts/Util.py", line 1729, in runWithDriver
    current.driver.runClientServerTestCase(current)
  File "../scripts/LocalDriver.py", line 565, in runClientServerTestCase
    self.runner.runClientSide(client, current, host)
  File "../scripts/LocalDriver.py", line 197, in runClientSide
    testcase._runClientSide(current, host)
  File "../scripts/Util.py", line 1677, in _runClientSide
    self.runClientSide(current)
  File "../scripts/Util.py", line 1569, in runClientSide
    self._runClient(current, client)
  File "../scripts/Util.py", line 1689, in _runClient
    client.run(current)
  File "../scripts/Util.py", line 1206, in run
    self.stop(current, True, exitstatus)
  File "../scripts/Util.py", line 1278, in stop
    process.waitSuccess(exitstatus=exitstatus, timeout=30)
  File "../scripts/Util.py", line 2140, in waitSuccess
    raise RuntimeError("unexpected exit status: expected: %d, got %d\n" % (exitstatus, result))
RuntimeError: unexpected exit status: expected: 0, got 1
test in Ice/metrics failed:
Traceback (most recent call last):
  File "../scripts/Util.py", line 1700, in run
    self.runWithDriver(current)
  File "../scripts/Util.py", line 1729, in runWithDriver
    current.driver.runClientServerTestCase(current)
  File "../scripts/LocalDriver.py", line 565, in runClientServerTestCase
    self.runner.runClientSide(client, current, host)
  File "../scripts/LocalDriver.py", line 197, in runClientSide
    testcase._runClientSide(current, host)
  File "../scripts/Util.py", line 1677, in _runClientSide
    self.runClientSide(current)
  File "../scripts/Util.py", line 1569, in runClientSide
    self._runClient(current, client)
  File "../scripts/Util.py", line 1689, in _runClient
    client.run(current)
  File "../scripts/Util.py", line 1206, in run
    self.stop(current, True, exitstatus)
  File "../scripts/Util.py", line 1278, in stop
    process.waitSuccess(exitstatus=exitstatus, timeout=30)
  File "../scripts/Util.py", line 2140, in waitSuccess
    raise RuntimeError("unexpected exit status: expected: %d, got %d\n" % (exitstatus, result))
RuntimeError: unexpected exit status: expected: 0, got 1
Traceback (most recent call last):
  File "../scripts/LocalDriver.py", line 501, in runTestSuite
    testcase.run(current)
  File "../scripts/Util.py", line 1700, in run
    self.runWithDriver(current)
  File "../scripts/Util.py", line 1729, in runWithDriver
    current.driver.runClientServerTestCase(current)
  File "../scripts/LocalDriver.py", line 565, in runClientServerTestCase
    self.runner.runClientSide(client, current, host)
  File "../scripts/LocalDriver.py", line 197, in runClientSide
    testcase._runClientSide(current, host)
  File "../scripts/Util.py", line 1677, in _runClientSide
    self.runClientSide(current)
  File "../scripts/Util.py", line 1569, in runClientSide
    self._runClient(current, client)
  File "../scripts/Util.py", line 1689, in _runClient
    client.run(current)
  File "../scripts/Util.py", line 1206, in run
    self.stop(current, True, exitstatus)
  File "../scripts/Util.py", line 1278, in stop
    process.waitSuccess(exitstatus=exitstatus, timeout=30)
  File "../scripts/Util.py", line 2140, in waitSuccess
    raise RuntimeError("unexpected exit status: expected: %d, got %d\n" % (exitstatus, result))
RuntimeError: unexpected exit status: expected: 0, got 1
```

**Comment by bentoi** [comment]
I wasn't able to reproduce. I committed a fix to figure which value is wrong. Let's reopen when this shows up again.

**Comment by bentoi** [comment]
It failed with again with the latest Ice CI job. Still on iOS but with C++ this time:
```
(executing `Ice/metrics/server' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.InstanceName=Server', '--Ice.NullHandleAbort=1', '--Ice.ThreadPool.Server.SizeMax=3', '--Ice.ThreadPool.Server.Size=1', '--Ice.Warn.Connections=1', '--Ice.ThreadPool.Server.SizeWarn=0', '--IceSSL.CertFile=server.p12', '--IceSSL.KeychainPassword=password', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice/3.7/certs', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--Ice.PrintStackTraces=1', '--IceSSL.Password=password', '--IceSSL.Keychain=server.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.CAs=cacert.pem'])
(executing `Ice/metrics/client' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Warn.Connections=1', '--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceSSL.Password=password', '--IceSSL.Keychain=client.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Admin.InstanceName=Client', '--Ice.NullHandleAbort=1', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--IceSSL.CAs=cacert.pem', '--IceSSL.CertFile=client.p12', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice/3.7/certs', '--IceSSL.KeychainPassword=password', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--Ice.PrintStackTraces=1'])
testing metrics admin facet checkedCast... ok
testing group by none...ok
testing group by id...ok
testing connection metrics... ok
testing connection establishment metrics... ok
testing dispatch metrics... ok
testing invocation metrics... failed!
test/Ice/metrics/AllTests.cpp:1329: assertion `rim1->total == 6' failed
unexpected unknown exception while running `Ice/metrics client'

(executing `Ice/metrics/server' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.InstanceName=Server', '--Ice.NullHandleAbort=1', '--Ice.ThreadPool.Server.SizeMax=3', '--Ice.ThreadPool.Server.Size=1', '--Ice.Warn.Connections=1', '--Ice.ThreadPool.Server.SizeWarn=0', '--IceSSL.CertFile=server.p12', '--IceSSL.KeychainPassword=password', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice/3.7/certs', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--Ice.PrintStackTraces=1', '--IceSSL.Password=password', '--IceSSL.Keychain=server.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.CAs=cacert.pem'])
(executing `Ice/metrics/client' on `iOS Simulator (iOS-13-0)' args = ['--Ice.Warn.Connections=1', '--Ice.Plugin.IceSSL=IceSSL:createIceSSL', '--IceSSL.Password=password', '--IceSSL.Keychain=client.keychain', '--Ice.ThreadPool.Server.Serialize=1', '--Ice.Admin.InstanceName=Client', '--Ice.NullHandleAbort=1', '--Ice.Default.Protocol=ssl', '--Ice.Default.Host=127.0.0.1', '--Ice.IPv6=0', '--IceMX.Metrics.Parent.GroupBy=parent', '--IceSSL.CAs=cacert.pem', '--IceSSL.CertFile=client.p12', '--IceMX.Metrics.Debug.GroupBy=id', '--IceSSL.DefaultDir=/Users/vagrant/workspace/ice/3.7/certs', '--IceSSL.KeychainPassword=password', '--IceMX.Metrics.All.GroupBy=none', '--Ice.Admin.Endpoints=tcp -h 127.0.0.1', '--Ice.PrintStackTraces=1'])
testing metrics admin facet checkedCast... ok
testing group by none...ok
testing group by id...ok
testing connection metrics... ok
testing connection establishment metrics... ok
testing dispatch metrics... ok
testing invocation metrics... failed!
test/Ice/metrics/AllTests.cpp:1329: assertion `rim1->total == 6' failed
unexpected unknown exception while running `Ice/metrics client'
```

**Comment by pepone** [comment]
I pushed a few extra fixes for the test 4f7100d5

**Comment by pepone** [comment]
let's reopen when it happens again, didn't fail since Benoit added the extra diagnostic
